src/main/java/com/iexec/worker/docker/DockerService.java

```diff
@@ -202,6 +187,80 @@ public String getIexecOutBind(String chainTaskId) {
                 IexecFileHelper.SLASH_IEXEC_OUT;
     }
 
+    /**
+     * Stop all running containers launched by the worker via this current service
+     * and remove them from running containers record. The container itself is not
+     * removed here as it is removed by its watcher thread.
+     */
+    public void stopAllRunningContainers() {
+        log.info("About to stop all running containers [runningContainers:{}]",
+                runningContainersRecord);
+        List.copyOf(runningContainersRecord)
+                .forEach(containerName -> stopRunningContainer(containerName));
+    }
+
+    /**
+     * Stop running containers related to a certain task and remove them from the
+     * running containers record. The container can be pre-compute, compute, or
+     * post-compute. This is needed when the worker aborts a task. The container itself
+     * is not removed here as it is removed by its watcher thread.
+     * 
+     * @param chainTaskId
+     */
+    public void stopTaskRunningContainers(String chainTaskId) {
+        log.info("Cleaning task containers [chainTaskId:{}]", chainTaskId);
+        runningContainersRecord.stream()
+                .filter(containerName -> containerName.contains(chainTaskId))
+                .forEach(containerName -> stopRunningContainer(containerName));
+    }
+
+    /**
+     * Stop an running container with the provided containerName and remove it from
+     * running containers record. The container itself is not stopped here as it is
+     * removed by its watcher thread.
+     * 
+     * @param containerName
+     */
+    void stopRunningContainer(String containerName) {
+        if (!getClient().isContainerPresent(containerName)) {
+            log.error("No running container to be removed [containerName:{}]", containerName);
+            return;
+        }
+        if (!getClient().isContainerActive(containerName)) {
+            log.info("Container is not active it will be removed by its watcher thread"
+                    + "[containerName:{}]", containerName);
+        } else if (!getClient().stopContainer(containerName)) {
+            log.error("Failed to stop running container [containerName:{}]", containerName);
+            // Don't remove from record.
+            return;
+        }
+        removeFromRunningContainersRecord(containerName);
+    }
+
+    /**
+     * Get the record of running containers. Added originally for testing purposes.
+     * 
+     * @return
+     */
+    HashSet<String> getRunningContainersRecord() {
+        return runningContainersRecord;
+    }
+
+    /**
+     * Add a container to the running containers record
+     *
+     * @param containerName name of the container to be added to the record
+     * @return true if container is added to the record
+     */
+    boolean addToRunningContainersRecord(String containerName) {
+        if (runningContainersRecord.contains(containerName)) {
```

Comment:
Should not be a problem in current methods using it, but it is safer to:
`if containerName is empty or record contains`